### PR TITLE
refactor: fix golint issues

### PR DESCRIPTION
**Pull request by scortier**
### Description

- Fix the linting issues identified using linting tool.
- Add the linting tool `golangci-lint` in CI through Github Action

### Related Issues
Issue #205

### CODE REVIEW

File: plugins/extractors/date/date.go [hunk 1]
```diff
@@ -17,13 +17,16 @@ import (
 var summary string
 
 const (
+	// MaxDates is the maximum number of dates to extract
 	MaxDates = 3
```
Comment (ravisuhag) [inline plugins/extractors/date/date.go:21]: @scortier this can be a private variable.

File: plugins/extractors/mssql/mssql.go [hunk 2]
```diff
@@ -38,20 +39,23 @@ var sampleConfig = `
  user_id: admin
  password: "1234"`
 
+// Extractor manages the extraction of data from the database
 type Extractor struct {
 	out         chan<- interface{}
 	excludedDbs map[string]bool
 
-	// depedencies
+	// dependencies
```
Comment (ravisuhag) [inline plugins/extractors/mssql/mssql.go:47]: @scortier We can avoid this comment.

File: agent/agent_test.go [hunk 3]
```diff
@@ -117,7 +126,10 @@ func TestRunnerRun(t *testing.T) {
 	t.Run("should record metrics", func(t *testing.T) {
 		mSink := new(mockPassthroughSink)
 		sinkFactory := registry.NewSinkFactory()
-		sinkFactory.Register("mock-sink", newMockSinkFn(mSink))
+		err := sinkFactory.Register("mock-sink", newMockSinkFn(mSink))
```
Comment (ravisuhag) [inline agent/agent_test.go:129]: @scortier all such pattern can combined like this. 
if err := sinkFactory.Register("mock-sink", newMockSinkFn(mSink)) err != nil {
t.Error(err.Error())
}

File: test/dockertest.go [hunk 4]
```diff
@@ -6,19 +6,19 @@ import (
...
 )
 
-//This will create a docker container using the RunOptions given
+//CreateContainer will create a docker container using the RunOptions given
 //
 //"opts" is the configuration for docker
 //
 //"retryOp" is an exponential backoff-retry, because the application in the container might not be ready to accept connections yet
-func CreateContainer(opts dockertest.RunOptions, retryOp func(r *dockertest.Resource) error) (err error, purgeFn func() error) {
+func CreateContainer(opts dockertest.RunOptions, retryOp func(r *dockertest.Resource) error) (purgeFn func() error, err error) {
```
Comment (ravisuhag) [inline test/dockertest.go:14]: @scortier All integration tests needs to be refactored for this. You can try that by running with integration tag. 
cc @GrayFlash

Comment (GrayFlash) [reply test/dockertest.go:14]: yup, so @scortier we need to go through all the tests in plugins/extractors and interchange position of output variables this function returns, and test it using `go test -tags=integration`
for e.g.;
err, purgeFn := 
becomes
purgefn, err :=

Comment (scortier) [reply test/dockertest.go:14]: got it 👍

File: agent/agent.go [hunk 5]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/pkg/errors"
 )
 
+// Agent contains fields responsible for the execution of a recipe.
```
Comment (ravisuhag) [inline agent/agent.go:16]:
```suggestion
// Agent runs recipes for specified plugins.
```

File: agent/agent.go [hunk 6]
```diff
@@ -21,6 +22,7 @@ type Agent struct {
 	logger           log.Logger
 }
 
+// NewAgent creates a new Agent.
```
Comment (ravisuhag) [inline agent/agent.go:25]:
```suggestion
// NewAgent returns an Agent with plugin factories.
```

File: agent/agent.go [hunk 7]
```diff
@@ -34,13 +36,12 @@ func NewAgent(ef *registry.ExtractorFactory, pf *registry.ProcessorFactory, sf *
 	}
 }
 
+// Validate validates the recipe.
```
Comment (ravisuhag) [inline agent/agent.go:39]:
```suggestion
// Validate checks the recipe for linting errors.
```

File: agent/agent.go [hunk 8]
```diff
@@ -63,14 +63,14 @@ func (r *Agent) Validate(rcp recipe.Recipe) (errs []error) {
...
-		err = procc.Validate(p.Config)
-		if err != nil {
+		if err = procc.Validate(p.Config); err != nil {
 			errs = append(errs, errors.Wrapf(err, "invalid config for %s (%s)", p.Name, plugins.PluginTypeProcessor))
 		}
 	}
 	return
 }
 
+// RunMultiple runs multiple recipes.
```
Comment (ravisuhag) [inline agent/agent.go:73]:
```suggestion
// RunMultiple executes multiple recipes.
```

File: agent/agent.go [hunk 9]
```diff
@@ -92,6 +92,7 @@ func (r *Agent) RunMultiple(recipes []recipe.Recipe) []Run {
 	return runs
 }
 
+// Run runs a recipe.
```
Comment (ravisuhag) [inline agent/agent.go:95]:
```suggestion
// Run executes the specified recipe.
```

File: agent/run.go [hunk 10]
```diff
@@ -2,14 +2,19 @@ package agent
 
 import "github.com/odpf/meteor/recipe"
 
+// TaskType is the type of a task
 type TaskType string
 
 const (
+	// TaskTypeExtract is the type of a task that extracts a file
```
Comment (ravisuhag) [inline agent/run.go:9]:
```suggestion
	// TaskTypeExtract is the type of task that extracts a record
```

File: agent/run.go [hunk 11]
```diff
@@ -2,14 +2,19 @@ package agent
...
 	TaskTypeExtract TaskType = "extract"
+	// TaskTypeProcess is the type of a task that processes a file
```
Comment (ravisuhag) [inline agent/run.go:11]:
```suggestion
	// TaskTypeProcess is the type of task that processes a record
```

File: agent/run.go [hunk 12]
```diff
@@ -2,14 +2,19 @@ package agent
...
 	TaskTypeProcess TaskType = "process"
+	// TaskTypeSink is the type of a task that sends a file to a sink
```
Comment (ravisuhag) [inline agent/run.go:13]:
```suggestion
	// TaskTypeSink is the type of task that sends a record to a sink
```

File: generator/recipe.go [hunk 13]
```diff
@@ -14,13 +14,15 @@ var file embed.FS
 
 var size uint = 2
 
+// Template contains the set of fields that are used to generate the recipe
```
Comment (ravisuhag) [inline generator/recipe.go:17]:
```suggestion
// Template represents the template for generating a recipe.
```

File: metrics/statsd.go [hunk 14]
```diff
@@ -14,18 +14,21 @@ var (
 	runMetricName         = "run"
 )
 
+// StatsdMonitor contains the statsd client and prefix
```
Comment (ravisuhag) [inline metrics/statsd.go:17]:
```suggestion
// StatsdMonitor reprsents the statsd monitor.
```

**Review by ravisuhag — CHANGES_REQUESTED** [review]

**Pull request merged**